File: 代码/kyzh_parent/kyzh_manager_web/src/main/webapp/js/service/dataSortingService.js
//定义brandService，前端服务，发送ajax请求
app.service('dataSortingService',function ($http) {

    this.findPage = function (pageNum,pageSize,searchEntity) {
        return $http.post('../dataSorting/findPage.do?pageNum='+pageNum+"&pageSize="+pageSize,searchEntity);
    }

    this.insert = function (entity) {
        return $http.post('../dataSorting/insert.do',entity);
    }

    this.update = function (entity) {
        return $http.post('../dataSorting/update.do',entity);
    }

    this.findOne = function (id) {
        return $http.get('../dataSorting/findOne.do?id='+id);
    }

    this.del = function (ids) {
        return $http.get('../dataSorting/delete.do?ids='+ids);
    }

    this.findByParentId = function (id) {
        return $http.get('../dataSorting/findByParentId.do?id='+id);
    }
})

File: 代码/kyzh_parent/kyzh_article_interface/src/main/java/com/kyzh/article/service/DataSortingService.java
package com.kyzh.article.service;

import com.kyzh.pojo.TbDataSorting;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 资料分类
 */
public interface DataSortingService {

    List<TbDataSorting> findAll();

    PageResult findPage(int pageNum, int pageSize,TbDataSorting dataSorting);

    void insert(TbDataSorting dataSorting);

    TbDataSorting findOne(Long id);

    void update(TbDataSorting dataSorting);

    void delete(Long[] ids);

    List<Map> selectOptions();

    /**
     * 根据id查询该分类下的所有分类
     * @param id
     * @return
     */
    List<TbDataSorting> findByParentId(Long id);
}
